ElectronApp/src/renderer/components/EditableTable.tsx

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint react/prop-types: 0 */
+
+import React from 'react';
+import styled from 'styled-components';
+import { useTable, usePagination } from 'react-table';
+
+import makeData from './makeData';
+
+declare module 'react-table' {
+  interface TableOptions<D extends object> {
+    updateMyData: (rowIndex: number, columnId: string, value: any) => void;
+  }
+}
+
+const Styles = styled.div`
+  padding: 1rem;
+
+  table {
+    border-spacing: 0;
+    border: 1px solid black;
+
+    tr {
+      :last-child {
+        td {
+          border-bottom: 0;
+        }
+      }
+    }
+
+    th,
+    td {
+      margin: 0;
+      padding: 0.5rem;
+      border-bottom: 1px solid black;
+      border-right: 1px solid black;
+
+      :last-child {
+        border-right: 0;
+      }
+
+      input {
+        font-size: 1rem;
+        padding: 0;
+        margin: 0;
+        border: 0;
+      }
+    }
+  }
+`;
+
+export interface EditableCellProps {
+  value: any;
```

Comment:
I know we struggled with adding types here, but let's try to make it work! See if we can get rid of all the `any`'s here. There are types provided by the library types/react-table. For example:

```
export interface EditableCellProps {
  value: CellValue;
  row: Row;
  column: Column;
  updateMyData: Function;
}
```

Reply:
Probably not for this table(because the interface is defined by someone else) but you can add "jsdoc/tsdoc" comments like below which turn up when you're hovering.

```tsx
export interface EditableCellProps { 
    /** Called after input is committed, not after every key press. */
    updateMyData: Function; 
}
```